src/app/pages/permissions-page/_services/permissions.model.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+import { AuthService } from '@savvato-software/savvato-javascript-services';
+import { PermissionsApiService } from './permissions.api.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PermissionsModelService{
+  
+  userModel: any = {};
+  
+  roleModel: any = {};
+
+  currentUserModel: any = {};
+
+
+  constructor(private _authService: AuthService,
+    private _permissionsApiService: PermissionsApiService) {}
+
+    initUser() {
```

Comment:
Generally, there should be just one init() method. The problem is that you have several api calls necessary to finish initialization... hmm, what to do? right? Actually, I think this would be a good thing to talk about tonight. The solution, if you want to take a look at it beforehand, is in the repo savvato-javascript-services. There is a class there called model-transforming.service. If you'd like take a look, and see if you can see how it would be applied. Regardless, I'll talk about it in the next meeting if you're available.